### PR TITLE
refac: rewrite charge types test

### PR DESCRIPTION
**Pull request by johevemi**
Rewrites test_charge_types to test everything through public functions instead of each private function. This is to make the code easier to refactor

### CODE REVIEW

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 1]
```diff
@@ -12,933 +12,353 @@
...
+DEFAULT_ENERGY_SUPPLIER_ID = "1234567890123"
+DEFAULT_METERING_POINT_ID = "123456789012345678901234567"
+DEFAULT_METERING_POINT_TYPE = MeteringPointType.CONSUMPTION
+DEFAULT_SETTLEMENT_METHOD = SettlementMethod.FLEX
+DEFAULT_QUANTITY = Decimal("1.005")
+DEFAULT_QUALITY = ChargeQuality.CALCULATED
+DEFAULT_PERIOD_START_DATETIME = datetime(2019, 12, 31, 23)
+
+
+def _create_metering_point_row(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:58]: question: are we going away from using the data factory methods?

Comment (JesperGomes) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:58]: ...same questions for the other create-methods

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 2]
```diff
@@ -12,933 +12,353 @@
...
     MeteringPointType,
     SettlementMethod,
-)
-from package.calculation.wholesale.schemas.charges_schema import (
-    charges_master_data_schema,
-)
-from package.calculation.preparation.transformations.charges_reader import (
-    _create_charges_df,
+    MeteringPointResolution,
+    TimeSeriesQuality,
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:34]: Idea: How about starting a trend like this?
```suggestion
import package.codelists as E
```
Where `E` would be short for "enum".

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 3]
```diff
@@ -12,933 +12,353 @@
...
-        (explode_charges_with_price_and_links_dataset_1, 31),
-        (explode_charges_with_price_and_links_dataset_2, 0),
-        (explode_charges_with_price_and_links_dataset_3, 2),
-        (explode_charges_with_price_and_links_dataset_4, 0),
-    ],
+    "charge_resolution", [ChargeResolution.HOUR, ChargeResolution.DAY]
 )
-def test__explode_subscription__explodes_into_rows_based_on_number_of_days_between_from_and_to_date(
-    spark: SparkSession, charges_df: DataFrame, expected: int
+def test__get_tariff_charges__filters_on_resolution_hour_or_day(
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:144]: How about?
```suggestion
def test__get_tariff_charges__filters_on_resolution(
```

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 4]
```diff
@@ -12,933 +12,353 @@
...
+    actual = get_tariff_charges(
+        metering_point,
+        time_series,
+        charges,
+        charge_resolution,
+    )
 
-    # Assert
-    assert result.count() == expected
+    assert actual.collect()[0][Colname.charge_resolution] == charge_resolution.value
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:173]: Could the other resolution be in a second row in the actual data frame?

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 5]
```diff
@@ -12,933 +12,353 @@
...
+    actual_subscription = get_subscription_charges(charges, metering_point)
+    assert actual_subscription.count() == 31
 
-@pytest.fixture(scope="session")
-def default_metering_point_period(
-    metering_point_period_factory: Callable[..., DataFrame]
-) -> DataFrame:
-    return metering_point_period_factory(DEFAULT_FROM_DATE, DEFAULT_TO_DATE)
 
+def test__get_tariff_charges__joins_on_metering_point_id_and_time_is_between_from_and_to_date(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:240]: not easy for me to understand what we are testing here. Do we need the word 'join' - it sounds like an implementation detail

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 6]
```diff
@@ -12,933 +12,353 @@
...
-        (
-            charges_with_price_and_links_dataset_1,
-            metering_points_dataset,
-            1,
-        ),
-        (
-            charges_with_price_and_links_dataset_2,
-            metering_points_dataset,
-            0,
+def test__get_charges_functions__filters_on_correct_charge_type(
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:176]: This test seems to be testing multiple suts. Leading to a generic name that doesn't either tell what the sut is.
I can see why you did this, but how about splitting the tests and make them more explicit and shorter each?

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 4 above)
Comment (johevemi) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:173]: no. I will add an assert on count

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:58]: if it is up to me then yes, I think this is easier

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 7]
```diff
@@ -12,933 +12,353 @@
...
     )
 
+    assert actual.count() == 2
+
 
-def test__get_tariff_charges__when_no_charge_data_match_the_resolution__returns_empty_tariffs(
-    default_metering_point_period: DataFrame,
-    default_time_series_point: DataFrame,
-    charges_factory: Callable[..., DataFrame],
+def test__get_tariff_charges__sums_quantity_when_same_metering_point_and_resolution(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:290]: or:
test__get_tariff_charges__when_same_metering_point_and_resolution__sums_quantity

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:34]: good idea

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 5 above)
Comment (johevemi) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:233]: what about test__get_tariff_charges__only_accepts_charges_in_metering_point_period

File: source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 6 above)
Comment (johevemi) [reply source/databricks/calculation-engine/tests/calculation/preparation/transformations/test_charge_types.py:169]: yeah, I was not sure about that one. I have split it into 3 tests

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**